### PR TITLE
after hard work with Mirco and Philipp we came up with some changes i…

### PR DESCRIPTION
**Pull request by dsidoren**
…n GribEncoder which are required to run with FESOM

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](https://bol-claassistant.ecmwf.int:443/ecmwf/multio?pullRequest=18) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bol-claassistant.ecmwf.int:443/ecmwf/multio?pullRequest=18) before we can accept your contribution.<br/><hr/>**Dmitry Sidorenko** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bol-claassistant.ecmwf.int:443/check/ecmwf/multio?pullRequest=18) it.</sub>

File: src/multio/action/encode/GribEncoder.cc [hunk 1]
```diff
@@ -360,13 +360,25 @@ void GribEncoder::setOceanMetadata(const message::Message& msg) {
     }
 
     // Set ocean grid information
-    setValue("unstructuredGridType", config_.getString("grid-type"));
+    if ( config_.has("grid-type") ){
```
Comment (dsarmany) [inline src/multio/action/encode/GribEncoder.cc:363]: We could also have the one-liner:
```c++
setValue("unstructuredGridType", config_.getString("grid-type", "undefined"));
```
That would set `grid-type` to `undefined` automatically if that key is not found in the configuration.

Comment (dsarmany) [inline src/multio/action/encode/GribEncoder.cc:363]: We could also have the one-liner:
```c++
setValue("unstructuredGridType", config_.getString("grid-type", "undefined"));
```
That would set `grid-type` to `undefined` automatically if that key is not found in the configuration.

File: src/multio/action/encode/GribEncoder.cc [hunk 2]
```diff
@@ -360,13 +360,25 @@ void GribEncoder::setOceanMetadata(const message::Message& msg) {
...
+       if ( config_.getString("grid-type") != "undefined" ) {
+               setValue("unstructuredGridType", config_.getString("grid-type"));
+       }
+    }
 
-    const auto& gridSubtype = metadata.getString("gridSubtype");
-    setValue("unstructuredGridSubtype", gridSubtype.substr(0, 1));
+     const auto& gridSubtype = metadata.getString("gridSubtype");
```
Comment (dsarmany) [inline src/multio/action/encode/GribEncoder.cc:369]: Similarly:
```c++
setValue("unstructuredGridSubtype", config_.getString("gridSubtype", "undefined")); // Or just "F" if "undefined" does not work
```

File: src/multio/action/encode/GribEncoder.cc [hunk 3]
```diff
@@ -360,13 +360,25 @@ void GribEncoder::setOceanMetadata(const message::Message& msg) {
...
+            setValue("unstructuredGridSubtype", gridSubtype.substr(0, 1));
+     }
 
-    const auto& gridUID = metadata.getString("uuidOfHGrid");
-    setValue("uuidOfHGrid", gridUID);
+//    setValue("unstructuredGridType", config_.getString("grid-type"));
+//    const auto& gridSubtype = metadata.getString("gridSubtype");
+//    setValue("unstructuredGridSubtype", gridSubtype.substr(0, 1));
+
+    if ( metadata.has("uuidOfHGrid") ){
```
Comment (dsarmany) [inline src/multio/action/encode/GribEncoder.cc:378]: And here:
```c++
setValue("uuidOfHGrid", config_.getString("uuidOfHGrid", "00000000000000000000000000000000"));
```

**Review by dsarmany — CHANGES_REQUESTED** [review]
I'm glad you've managed to get it work!

It would be even nicer if you could come up with some names for gridType and gridSubtype for the FESOM grids you may want to use and put it in the definition tables. Now that data governance have made the the heavy lifting of creating the framework, it is not much of an effort.

**Comment by geier1993** [comment]
@dsarmany I think we can close this now.

**Comment by dsarmany** [comment]
Close it as the it has been resolved in a slightly different branch/way.

**Pull request closed without merging**